### PR TITLE
[Merged by Bors] - feat(Ergodic): the left action of a group on itself is ergodic

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #14595
- [x] depends on: #14574
- [x] depends on: #14575
- [x] depends on: #14952

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Dynamics/Ergodic/Action/Regular.lean [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+In this file we prove that the left and right actions of a group on itself are ergodic.
+-/
+
+open MeasureTheory Measure Filter Set
+open scoped Pointwise
+
+variable {G : Type*} [Group G] [MeasurableSpace G] [MeasurableMul₂ G] [MeasurableInv G]
+  {μ : Measure G} [SFinite μ]
+
+instance [μ.IsMulLeftInvariant] : ErgodicSMul G G μ := by
```
Comment (ocfnash) [inline Mathlib/Dynamics/Ergodic/Action/Regular.lean:21]:
```suggestion
@[to_additive]
instance [μ.IsMulLeftInvariant] : ErgodicSMul G G μ := by
```

File: Mathlib/Dynamics/Ergodic/Action/Regular.lean [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  intro hμs
+  obtain ⟨a, has, ha⟩ : ∃ a ∈ s, ∀ᵐ b ∂μ, (b * a ∈ s ↔ a ∈ s) := by
+    refine (hμs.and_eventually ?_).exists
+    rw [ae_ae_comm]
+    · exact ae_of_all _ fun b ↦ (hs b).mem_iff
+    · exact ((hsm.preimage <| measurable_snd.mul measurable_fst).mem.iff
+        (hsm.preimage measurable_fst).mem).setOf
+  simpa [has] using (MeasureTheory.quasiMeasurePreserving_mul_right μ a⁻¹).ae ha
+
+instance [μ.IsMulRightInvariant] : ErgodicSMul Gᵐᵒᵖ G μ := by
```
Comment (ocfnash) [inline Mathlib/Dynamics/Ergodic/Action/Regular.lean:35]:
```suggestion
@[to_additive]
instance [μ.IsMulRightInvariant] : ErgodicSMul Gᵐᵒᵖ G μ := by
```

**Review by ocfnash — APPROVED** [review]
It's possibly a slight shame not to factor the proof for the right action through that for the left. Then again, maybe it gets messy.

In any case,
bors d+

**Comment by urkud** [comment]
> It's possibly a slight shame not to factor the proof for the right action through that for the left. Then again, maybe it gets messy.

It needs a few missing instances. I added it to my TODO list.
bors merge

**Pull request closed without merging**